jaraco/classes/properties.py

```diff
@@ -1,3 +1,28 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar, cast, overload
+
+_T = TypeVar("_T")
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import Any, Protocol
+
+    from typing_extensions import Self, TypeAlias
```

Comment:
Did we determine that this dependency doesn't need to be declared? I think it should be declared in setup.cfg for `tests` extras (presumably unconditionally on Python version as it's imported unconditionally).

Reply:
`typing_extensions` is a hard dependency of `mypy` (https://github.com/python/mypy/blob/780a29d0b1b93848d7ce3faf8e819a6cc140b2e5/setup.py#L220-L225) and we might be sure it won't go away. Fortunately, we would only need to add it to the dependency specs if it was imported at runtime.

Reply:
And because we use `TYPE_CHECKING`, no mypy = no need of `typing_extensions`. So yeah, no need to decouple these.